jax/numpy/lax_numpy.py

```diff
@@ -3237,6 +3237,11 @@ def sort(a, axis=-1, kind='quicksort', order=None):
   else:
     return lax.sort(a, dimension=_canonicalize_axis(axis, ndim(a)))
 
+@_wraps(np.sort_complex)
+def sort_complex(a):
+  a = lax.sort(a, dimension=0)
+  return lax.convert_element_type(a, result_type(a, complex64))
```

Comment:
I think if you change `complex64` to `complex_`, it will fix the type errors seen in the CI.

To test it locally, you can run with x64 mode turned on (See https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#Double-(64bit)-precision)

Here's one invocation that will do this:
```
$ python tests/lax_numpy_test.py --num_generated_cases=50 --jax_enable_x64 --test_targets=testSortComplex
```

Reply:
Side-note: the reason this didn't come up before is because we had a mistake in our CI specifications, so no tests were being run in x64 mode! We fixed this in #3854